application/frontend/controllers/MethodController.php

```diff
@@ -38,6 +42,159 @@ public function actionIndex()
     {
         /** @var User $user */
         $user = \Yii::$app->user->identity;
-        return $user->methods;
+        return $user->getVerifiedMethods();
+    }
+
+    /**
+     * View single method
+     * @param string $uid
+     * @return Method
+     * @throws NotFoundHttpException
+     */
+    public function actionView($uid)
+    {
+        $method = Method::findOne(['uid' => $uid, 'user_id' => \Yii::$app->user->getId()]);
+        if ($method === null) {
+            throw new NotFoundHttpException();
+        }
+
+        return $method;
+    }
+
+    /**
+     * Create new unverified method and send verification
+     * @return Method
+     * @throws BadRequestHttpException
+     * @throws \Exception
+     */
+    public function actionCreate()
+    {
+        $request = \Yii::$app->request;
+
+        $type = $request->post('type');
+        $value = $request->post('value');
+
+        if ($type === null || ! in_array($type, [Method::TYPE_EMAIL, Method::TYPE_PHONE])) {
+            throw new BadRequestHttpException(
+                sprintf('Type is required. Options are: %s or %s', Method::TYPE_EMAIL, Method::TYPE_PHONE)
+            );
+        } elseif ($value === null) {
+            throw new BadRequestHttpException('Value is required.');
+        }
+
+        /*
+         * Check for existing method with this value
+         */
+        $method = Method::findOne(['value' => $value, 'user_id' => \Yii::$app->user->getId()]);
+        if ($method !== null) {
+            return $method;
+        }
+
+        /*
+         * Create method entry to be verified
+         */
+        $newMethod = Method::createAndSendVerification(
+            \Yii::$app->user->getId(),
+            $type,
+            $value
+        );
+
+        return $newMethod;
+    }
+
+    /**
+     * Validates user submitted code and marks method as verified if valid
+     * @param string $uid
+     * @return Method
+     * @throws BadRequestHttpException
+     * @throws NotFoundHttpException
+     * @throws TooManyRequestsHttpException
+     * @throws \Exception
+     */
+    public function actionUpdate($uid)
+    {
+        /*
+         * Delete methods not yet verified that have expired - Just-in-time cleanup
+         */
+        try {
+            Method::deleteExpiredUnverifiedMethods();
+        } catch (\Exception $e) {
+            \Yii::error([
+                'action' => 'deleteExpiredUnverifiedMethods',
+                'status' => 'error',
+                'error' => $e->getMessage(),
+            ]);
+        }
+
+        /*
+         * Find method if belongs to user and is not expired
+         */
+        /** @var Method $method */
+        $method = Method::findOne([
+            'uid' => $uid,
+            'user_id' => \Yii::$app->user->getId(),
+        ]);
+
+        /*
+         * If not found, throw 404
+         */
+        if ($method === null) {
+            throw new NotFoundHttpException();
```

Comment:
Do you want to throw the same exception for unfound method and invalid code, so that a hacker doesn't know when he has a good method.